emailjs/actions/send.ts

```diff
@@ -0,0 +1,40 @@
+import { AppContext } from "../mod.ts";
+import type { CreateEmailResponse } from "../utils/types.ts";
+
+export interface Props {
+  /**
+   * @description Template ID of the email
+   */
+  template_id: string;
+  /**
+   * @description Template parameters of the template
+   */
+  template_params?: { [key: string]: string };
```

Comment:
Did you test it using our admin?

It seems something that shouldn't work.. O_O

Usually we would do something like: Array<{ name: string, prop: value}>.

If the keys are previously known, it would be better to provide them here.

Reply:
> Did you test it using our admin?
> 
> It seems something that shouldn't work.. O_O
> 
> Usually we would do something like: Array<{ name: string, prop: value}>.
> 
> If the keys are previously known, it would be better to provide them here.

Yes, I actually tested it and found out that this was why template_params was not showing up in the Admin preview. Thanks for the observation, Matheus, I'll upload the fix soon. The keys can vary according to the user's choice (he's the one who defines them in the emailjs template, so maybe the best option is to make it variable here too)

Reply:
@matheusgr

I made a change that fixes the incompatibility between the Deco.cx admin format and the EmailJS API format for template parameters. The admin requires an array of objects with 'name' and 'value' keys, while the API expects an object with dynamic keys. The `templateParamsObject` function converts the array to the required object format using `reduce`, ensuring API compatibility before making the call.

```ts
const templateParamsObject = payload.template_params?.reduce(
  (acc, { name, value }) => {
    acc[name] = value;
    return acc;
  },
  {} as { [key: string]: string }
);
```